### PR TITLE
refactor(nms): reorganise context providers

### LINKED ISSUE #13242: Reorganize context providers
Today `LteContext` and `FegContext` render different context providers, hold the state for these contexts and implement update methods. The context are created in separate files under `/context` and the update methods use helper functions defined in `/state`.

This should be conciliated. Every context should have a separate file under `/context` it should hold:
- The context type definitions
- The context object created by `React.createContext`
- A component that renders the context provider, holds the state and implements the update methods.
- The helper functions that are used to fetch and update state

### PR DESCRIPTION
**Pull request by Neudrino**
## Summary

Moves all logic related to a given context type into one place.

- Fixes https://github.com/magma/magma/issues/13242

## Test Plan

`yarn tsc && yarn eslint . && yarn test` and
click through all tabs and make sure that there is not error.

### CODE REVIEW

File: nms/app/views/subscriber/__tests__/SubscriberAddEditTest.tsx [hunk 1]
```diff
@@ -464,15 +430,19 @@ describe('<AddSubscriberButton />', () => {
         },
       ],
     });
+
+    // await wait(undefined, {timeout: 42});
```
Comment (Neudrino) [inline nms/app/views/subscriber/__tests__/SubscriberAddEditTest.tsx]: Comment?

File: nms/app/views/equipment/__tests__/EquipmentGatewayPoolTest.tsx [hunk 2]
```diff
@@ -13,87 +13,75 @@
...
+    gateway_ids: [],
+    gateway_pool_id: 'pool3',
+    gateway_pool_name: 'pool_3',
   },
 };
+
+const lteNetworkIdGatewaysGatewayIdCellularPoolingGetMock = [
+  {
+    gateway_pool_id: 'pool1',
+    //gateway_id: 'gw1',
```
Comment (Neudrino) [inline nms/app/views/equipment/__tests__/EquipmentGatewayPoolTest.tsx]: Maybe keep the `gw1` etc but format nicely?

File: nms/app/components/context/SubscriberContext.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+};
+
+/* SubscriberContextType
+state: paginated subscribers
+sessionState: paginated subscribers session state
+metrics: subscriber metrics
+gwSubscriberMap: gateway subscriber map
+totalCount: total count of subscribers
+setState: POST, PUT, DELETE subscriber
+*/
```
Comment (Neudrino) [inline nms/app/components/context/SubscriberContext.tsx]: Could we style as `/**` comment or remove?

**Review by Neudrino — COMMENTED** [review]
:heavy_check_mark: Reviewed commits by @thmsschmitt . LGTM!

**Review by thmsschmitt — APPROVED** [review]
I had a look at the changes from @Neudrino also LGTM.

**Pull request merged**